Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RenderPass.cpp

```diff
@@ -193,6 +193,12 @@ namespace AZ
         void RenderPass::FrameBeginInternal(FramePrepareParams params)
         {
             m_timestampResult = AZ::RPI::TimestampResult();
+
+            if (m_deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                m_deviceIndex = azrtti_cast<AZ::RPI::ParentPass*>(m_parent)->GetDeviceIndex();
```

Comment:
Do we need to do this check and set the value each frame? Are we expecting the device index to change after the RenderPass is initialized?

Reply:
I suppose we are expecting a pass can be re-allocated to different device dynamically in runtime? so I am doing it in each frame.

Reply:
Yes, same reason as for the `ParentPass`. I.e., the value should not be modified here either and taken from the parent if available when the scope is created. I would say you can move the `m_deviceIndex` to the `Pass` class even, since there are only two more passes that derive directly from it rather than the `RenderPass` or `ParentPass` and these are `ImageAttachmentPreviewPass` and `SelectorPass` and it they can just ignore it if they don't need it.

Reply:
That makes sense, I've changed it as requested